rts/Sim/Path/HAPFS/PathDataTypes.h

```diff
@@ -39,7 +39,14 @@ struct PathNode {
 /// functor to define node priority
 struct lessCost {
 	inline bool operator() (const PathNode* x, const PathNode* y) const {
-		return (x->fCost == y->fCost) ? (x->gCost < y->gCost) : (x->fCost > y->fCost);
+		if (x->fCost == y->fCost) {
+			if (x->gCost == y->gCost)
+				// This is needed to guarantee that the sorting is stable.
+				return (x->nodeNum < y->nodeNum);
+			else
+				return (x->gCost < y->gCost);
+		} else
+			return (x->fCost > y->fCost);
```

Comment:
Do `gCost` and `fCost` really have opposite meanings? Talking about:
* return (x->gCost < y->gCost);
* (x->fCost > y->fCost);

Reply:
In any case I'd rely on `std::tuple` comparison operator just to reduce the complexity and possible errors. Something like:
`return std::tie(y->fCost, x->gCost, x->nodeNum) < std::tie(x->fCost, y->gCost, y->nodeNum);`

Reply:
I really can't think why the gcost would be evaluated in the opposite direction to fcost. It would make sense to prioritize the nodes that are on the shortest route. std::tie is a good call.

Reply:
Sure, I was not aware of semantic meaning of `f` and `g`, just noticed they have opposite direction and suspected a possible typo. Good to know it's as intended.